internal/manager/conditions_crd_test.go

```diff
@@ -0,0 +1,75 @@
+package manager_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/manager"
+)
+
+func TestCRDControllerCondition(t *testing.T) {
+	knownGvr := schema.GroupVersionResource{
+		Group:    "group",
+		Version:  "v1",
+		Resource: "resources",
+	}
+	unknownGVR := schema.GroupVersionResource{
+		Group:    "otherGroup",
+		Version:  "v1",
+		Resource: "resources",
+	}
+
+	restMapper := meta.NewDefaultRESTMapper(nil)
+	restMapper.Add(schema.GroupVersionKind{
+		Group:   knownGvr.Group,
+		Version: knownGvr.Version,
+		Kind:    "Resource",
+	}, meta.RESTScopeRoot)
+
+	testCases := []struct {
+		name           string
+		gvr            schema.GroupVersionResource
+		enabled        bool
+		expectedResult bool
+	}{
+		{
+			name:           "enabled_and_registered_resource",
+			gvr:            knownGvr,
+			enabled:        true,
+			expectedResult: true,
+		},
+		{
+			name:           "disabled_and_registered_resource",
+			gvr:            knownGvr,
+			enabled:        false,
+			expectedResult: false,
+		},
+		{
+			name:           "enabled_and_not_registered_resource",
+			gvr:            unknownGVR,
+			enabled:        true,
+			expectedResult: false,
+		},
+		{
+			name:           "disabled_and_not_registered_resource",
+			gvr:            unknownGVR,
+			enabled:        false,
+			expectedResult: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
What's the run time of those tests? If it's really fast (e.g. < 0.1s) I wouldn't bother using `.Parallel()` to get cleaner test logs.

Reply:
Yeah, they're much more under 0.1s: [f4584b3](https://github.com/Kong/kubernetes-ingress-controller/pull/3013/commits/f4584b309593c92827e9c3adea19e9f020765591)